src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AnalysisSections.kt

```diff
@@ -182,23 +185,198 @@ internal inline fun <R> runOutsideIgnoredSection(block: () -> R): R {
     }
 }
 
-internal fun getDefaultSilentSectionType(className: String, methodName: String): AnalysisSectionType? {
-    if (className.startsWith("java.util.concurrent.")) {
-        if (isJavaExecutorService(className)) {
-            if (methodName == "submit") {
-                return AnalysisSectionType.SILENT_PROPAGATING
+/**
+ * Configures how different code sections should be analyzed.
+ * Used to control which classes should be transformed for analysis, what analysis sections
+ * should be applied, and which methods should be hidden from trace results.
+ *
+ * @property analyzeStdLib Controls whether standard library code should be analyzed. When false,
+ *                        standard collections and concurrent collections are hidden,
+ *                        concurrent collections are muted.
+ */
+internal class AnalysisProfile(val analyzeStdLib: Boolean) {
+    
+    constructor(testConfiguration: ManagedCTestConfiguration?) : this(
+        testConfiguration is ModelCheckingCTestConfiguration && testConfiguration.stdLibAnalysisEnabled
+    )
+
+
+    /**
+     * Determines whether a given class and method should be transformed (instrumented) for analysis.
+     *
+     * @param className The fully qualified name of the class to check
+     * @param methodName The name of the method to check
+     * @return true if the class/method should be transformed, false otherwise
+     */
+    @Suppress("UNUSED_PARAMETER") // methodName is here for uniformity and might become useful in the future  
+    fun shouldTransform(className: String, methodName: String): Boolean {
+        // We do not need to instrument most standard Java classes.
+        // It is fine to inject the Lincheck analysis only into the
+        // `java.util.*` ones, ignored the known atomic constructs.
+        if (className.startsWith("java.")) {
+            if (className == "java.lang.Thread") return true
+            if (className.startsWith("java.util.concurrent.") && className.contains("Atomic")) return false
+            if (className.startsWith("java.util.")) return true
+            return false
+        }
+            if (className.startsWith("com.sun.")) return false
```

Comment:
Indentation is broken here and in other places in this function.

Reply:
Done